---
title: "RXFP3 Antagonist RNA-seq"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
    fig_width: 7
    fig_height: 7
theme: cosmo
---

Source: https://github.com/markziemann/rxfp3_ant

# Introduction

Izel Eraslan 189 Project Summary

Project 189 title / Aim: Determine whether the RXFP3 antagonist drug alter gene expression in mice brain that consumed high-fat diet.

Methodology:

* Male C57BL/6J mice were divided into two groups: (1) treatment / RXFP3-antagonist 2 (n=6), and (2) control / vehicle (n=6) and

* At the end of the experiment, RNA was isolation from five brain regions of interest from each mouse: hippocampal formation (HIP), hypothalamus (HYP), amygdala (AMG), periaqueductal gray (PAG) and nucleus incertus (NI).

* Total RNA’s were extracted using the Qiagen RNeasy Kit (250), (Cat# 74106, Hilden, Germany).

* Nanodrop, Qubit and Tapestation were used to complete the normalisation step.

* We will be comparing genes between the control vs drug within each brain areas.

Data form: FASTQ (see same distribution shown in the table below)

ID: from 105 – 164 only

hyp = hypothalamus = 1

hip = hippocampus = 2

amg = amygdala = 3

pag = periaqueductal gray = 4

ni = nucleus incertus = 5

| FileBasename | BrainRegion | Treatment | 
| --- | --- | --- |
| IE11-1 | hyp | T |
| IE12-1 | hyp | T |
| IE13-1 | hyp | T |
| IE14-1 | hyp | T |
| IE16-1 | hyp | T |
| IE20-1 | hyp | T |
| IE18-1 | hyp | C |
| IE19-1 | hyp | C |
| IE21-1 | hyp | C |
| IE22-1 | hyp | C |
| IE23-1 | hyp | C |
| IE24-1 | hyp | C |
| IE10-2 | hip | T |
| IE11-2 | hip | T |
| IE12-2 | hip | T |
| IE13-2 | hip | T |
| IE14-2 | hip | T |
| IE16-2 | hip | T | 
| IE17-2 | hip | C |
| IE19-2 | hip | C |
| IE21-2 | hip | C |
| IE22-2 | hip | C |
| IE23-2 | hip | C |
| IE24-2 | hip | C | 
| IE10-3 | amy | T |
| IE11-3 | amy | T |
| IE13-3 | amy | T |
| IE14-3 | amy | T |
| IE16-3 | amy | T |
| IE17-3 | amy | C |
| IE18-3 | amy | C |
| IE19-3 | amy | C |
| IE20-3 | amy | T |
| IE21-3 | amy | C |
| IE22-3 | amy | C |
| IE23-3 | amy | C |
| IE10-4 | pag | T |
| IE11-4 | pag | T |
| IE13-4 | pag | T |
| IE14-4 | pag | T |
| IE16-4 | pag | T |
| IE20-4 | pag | T |
| IE17-4 | pag | C |
| IE18-4 | pag | C |
| IE19-4 | pag | C |
| IE22-4 | pag | C |
| IE23-4 | pag | C |
| IE24-4 | pag | C |
| IE10-5 | ni | T |
| IE11-5 | ni | T |
| IE12-5 | ni | T |
| IE13-5 | ni | T |
| IE16-5 | ni | T |
| IE20-5 | ni | T |
| IE17-5 | ni | C |
| IE18-5 | ni | C |
| IE19-5 | ni | C |
| IE21-5 | ni | C |
| IE23-5 | ni | C |
| IE24-5 | ni | C |


## Bioinformatics methods

Fastqc (v0.11.9) was used to inspect sequence quality[1].

The mouse transcriptome was downloaded from GENCODE version 28[2].

Skewer (v0.2.2) was used to trim low quality bases (qual<20) from the 3' end of the read[3].

Kallisto (0.46.1) was used to map RNA-seq reads to the transcriptome [4].

Multiqc was used to tabulate sequence quality, trimming and mapping statistics [5].

Data were read into R v4.1.2 and duplicate lane data were aggregated, and transcript level counts were aggregated to gene level counts.

Genes with an average of less than 10 reads across samples were excluded from downstream analysis.

DESeq (1.32.0) was used with default settings to assay differential expression between control and treatment groups for all tissues [6].

Pathway analysis was performed with reactome gene sets obtained from MSigDB and converted to mouse gene identifiers with the msigdbr package 
(performed on 16-02-2022) [7,8,9].

Differential pathway analysis was performed with the "mitch" bioconductor package [10].

Genes and pathways with false discovery rate (FDR)<0.05 were considered significant.

```{r,packages}

suppressPackageStartupMessages({
    library("zoo")
    library("tidyverse")
    library("reshape2")
    library("DESeq2")
    library("gplots")
    library("fgsea")
    library("MASS")
    library("mitch")
    library("eulerr")
    library("limma")
    library("topconfects")
    library("kableExtra")
    library("vioplot")
    library("beeswarm")
})

```

## Import read counts

Importing RNA-seq data

```{r,importdata}

tmp <- read.table("3col.tsv.gz",header=F)
x <- as.matrix(acast(tmp, V2~V1, value.var="V3", fun.aggregate = sum))
x <- as.data.frame(x)
accession <- sapply((strsplit(rownames(x),"\\|")),"[[",2)
symbol<-sapply((strsplit(rownames(x),"\\|")),"[[",6)
x$geneid <- paste(accession,symbol)
xx <- aggregate(. ~ geneid,x,sum)
rownames(xx) <- xx$geneid
xx$geneid = NULL
xx <- round(xx)
xx <- xx[,which(colnames(xx)!="test")]
xx[1:6,1:6]
dim(xx)

```

Fix the sample names.

They are duplicated for lane 1 and 2, which I will aggregate.

```{r,colnames}

txx <- as.data.frame(t(xx))
txx$label <- sapply(strsplit(rownames(txx),"_"),"[[",1)
txx[1:3,c(1:4,ncol(txx))]
txx2 <- aggregate(. ~ label,txx,sum)
txx2[1:3,1:4]
rownames(txx2) <- txx2[,1]
txx2[,1] = NULL
xx <- as.data.frame(t(txx2))
xx[1:4,1:5]
write.table(xx,file="rxfp3_ant_counts.tsv",sep="\t",quote=FALSE)
rxx <- xx/colSums(xx) *1e6
rxx[1:4,1:5]

```

Samplesheet.

Need to delete "IE13-4".

```{r,ss}

ss <- read.table("rxfp3_ant_samplesheet.tsv",header=TRUE)
ss <- ss[order(ss$FileBasename),]
rownames(ss) <- ss$FileBasename
ss$FileBasename=NULL
ss <- ss[!rownames(ss)=="IE13-4",]
head(ss)

```

## QC analysis

Here I'll look at a few different quality control measures.

Firstly, the number of reads assigned to genes, which should be >15M.

```{r,qc1,fig.height=7,fig.width=7}

par(mar=c(5,8,3,1))
barplot(colSums(xx),horiz=TRUE,las=1,xlab="num reads",col=ss$cols)
sums <- colSums(xx)
sums <- sums[order(sums)]
barplot(sums,horiz=TRUE,las=1,xlab="num reads",cex.names=0.8)
abline(v=15000000,col="red")

```

Ribosomal RNA.

```{r,rrna}

rrna <- read.table("https://raw.githubusercontent.com/markziemann/rxfp3_ant/main/rrna_res.tsv")

myrrna <- rrna$V2/10000
names(myrrna) <- rrna$V1

barplot(myrrna,horiz=TRUE,las=1,xlab="percent rRNA reads",cex.names=0.8)

barplot(head(myrrna,10),horiz=TRUE,las=1,xlab="percent rRNA reads",cex.names=1)

```

This work identified IE10-5 had a high degree of rRNA reads, but that other IE10
samples also had an elevated rate of rRNA reads.

It was agreed to remove these.

## MDS plot for all samples

Multidimensional scaling plot to show the variation between all samples, very similar to PCA.

```{r,mds1,fig.height=7,fig.width=7}

mds <- cmdscale(dist(t(xx)))

cols <- as.numeric(as.factor(ss$BrainRegion))+1
pchs <- as.numeric(factor(ss$Treatment))+17

plot(mds, xlab="Coordinate 1", ylab="Coordinate 2",
  type = "p",bty="n",pch=pchs, cex=4 ,col=cols )
text(mds, labels=rownames(mds) ,col="black")

legend("bottomleft", inset=.02, title="tissue",
   legend=unique(as.factor(ss$BrainRegion)) , fill=unique(cols),  cex=1.2)

legend("left", inset=.02, title="treatment",
   legend=unique(as.factor(ss$Treatment)) , pch=unique(pchs),  cex=1.2)

```

Now would be a good time to remove IE10 from the samplesheet and matrix and re-generate
the MDS plot.

```{r,ie10_remove}

ss <- ss[grep("IE10",rownames(ss),invert=TRUE),]

xx <- xx[grep("IE10",colnames(xx),invert=TRUE),]

mds <- cmdscale(dist(t(xx)))

cols <- as.numeric(as.factor(ss$BrainRegion))+1
pchs <- as.numeric(factor(ss$Treatment))+17

plot(mds, xlab="Coordinate 1", ylab="Coordinate 2",
  type = "p",bty="n",pch=pchs, cex=4 ,col=cols )
text(mds, labels=rownames(mds) ,col="black")

legend("topleft", inset=.02, title="tissue",
   legend=unique(as.factor(ss$BrainRegion)) , fill=unique(cols),  cex=1.2)

legend("left", inset=.02, title="treatment",
   legend=unique(as.factor(ss$Treatment)) , pch=unique(pchs),  cex=1.2)

```

As you can see, the two MDS plots look completely different.

## Correlation heatmap

```{r,correl1}

heatmap.2(cor(xx),trace="n",main="Pearson correlation heatmap",margin=c(8,8),cexRow=0.7,cexCol=0.7)

```

## Set up the different datasets for differential expression analysis

Here, I'll give an example on how to separate the data matrix by tissue and then evaluate differential expression.

Don't forget to remove poorly detected genes from the matrix with a threshold of 10 reads per sample on average.

There are 5 contrasts to set up, one for each tissue.

The separate sample sheets are called s1, s2, etc.

The separate counts tables are called x1, x2, etc.

I will begin with hypothalamus and leave the rest to Craig's team.

```{r,filter}

dim(xx)
dim(ss)

ss1 <- ss[which(ss$BrainRegion=="hyp"),]
xx1 <- xx[which(colnames(xx) %in% rownames(ss1))]
xx1 <- xx1[which(rowMeans(xx1)>=10),]
rpm1 <- xx1/colSums(xx1) *1e6
dim(xx1)

ss2 <- ss[which(ss$BrainRegion=="hip"),]
xx2 <- xx[which(colnames(xx) %in% rownames(ss2))]
xx2 <- xx2[which(rowMeans(xx2)>=10),]
rpm2 <- xx2/colSums(xx2) *1e6
dim(xx2)

ss3 <- ss[which(ss$BrainRegion=="amy"),]
xx3 <- xx[which(colnames(xx) %in% rownames(ss3))]
xx3 <- xx3[which(rowMeans(xx3)>=10),]
rpm3 <- xx3/colSums(xx3) *1e6
dim(xx3)

ss4 <- ss[which(ss$BrainRegion=="pag"),]
xx4 <- xx[which(colnames(xx) %in% rownames(ss4))]
xx4 <- xx4[which(rowMeans(xx4)>=10),]
rpm4 <- xx4/colSums(xx4) *1e6
dim(xx4)

ss5 <- ss[which(ss$BrainRegion=="pag"),]
xx5 <- xx[which(colnames(xx) %in% rownames(ss5))]
xx5 <- xx5[which(rowMeans(xx5)>=10),]
rpm5 <- xx5/colSums(xx5) *1e6
dim(xx5)

```

## Differential expression with DESeq2

```{r,de01}

dds <- DESeqDataSetFromMatrix(countData = xx1 , colData = ss1, design = ~ Treatment )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:20,1:6] %>% kbl(caption = "Top gene expression differences for contrast 1: Effect of treatment in hypothalamus") %>% 
  kable_paper("hover", full_width = F)
dge1 <- dge
d1up <- rownames(subset(dge1,padj <= 0.05 & log2FoldChange > 0))
d1dn <- rownames(subset(dge1,padj <= 0.05 & log2FoldChange < 0))
write.table(dge1,file="dge1.tsv",quote=FALSE,sep="\t")

```

Here let's look at some plots.

MA plot shows the average level and fold change of all detected genes.
Volcano plot shows the fold change and the significance, as measured by -log(p-value).
Significant genes are shown as red points.

There are heatmaps of the top ranked genes by p-value.
Above the gene expression values there is a bar in orange/gray colours.
Control is shown in orange and treatment in grey.

```{r,deplots1}

maplot <- function(de,contrast_name) {
  de <- de[which(!is.na(de$padj)),]
  sig <-subset(de, padj < 0.05 )
  up <-rownames(subset(de, padj < 0.05 & log2FoldChange > 0))
  dn <-rownames(subset(de, padj < 0.05 & log2FoldChange < 0))
  GENESUP <- length(up)
  GENESDN <- length(dn)
  DET=nrow(de)
  SUBHEADER = paste(GENESUP, "up, ", GENESDN, "down", DET, "detected")
  ns <-subset(de, padj > 0.05 )
  plot(log2(de$baseMean),de$log2FoldChange,
       xlab="log2 basemean", ylab="log2 foldchange",
       pch=19, cex=0.5, col="dark gray",
       main=contrast_name, cex.main=1)
  points(log2(sig$baseMean),sig$log2FoldChange,
         pch=19, cex=0.5, col="red")
  mtext(SUBHEADER,cex = 1)
}

make_volcano <- function(de,name) {
    de <- de[which(!is.na(de$padj)),]
    de$pvalue[which(de$pvalue==0)] <- 1e-320
    sig <- subset(de,padj<0.05)
    N_SIG=nrow(sig)
    N_UP=nrow(subset(sig,log2FoldChange>0))
    N_DN=nrow(subset(sig,log2FoldChange<0))
    DET=nrow(de)
    HEADER=paste(N_SIG,"@5%FDR,", N_UP, "up", N_DN, "dn", DET, "detected")
    plot(de$log2FoldChange,-log10(de$pval),cex=0.5,pch=19,col="darkgray",
        main=name, xlab="log2 FC", ylab="-log10 pval")
    mtext(HEADER)
    grid()
    points(sig$log2FoldChange,-log10(sig$pval),cex=0.5,pch=19,col="red")
}

make_volcano2 <- function(de,name) {
    de <- de[which(!is.na(de$padj)),]
    de$pvalue[which(de$pvalue==0)] <- 1e-320
    sig <- subset(de,padj<0.05)
    N_SIG=nrow(sig)
    N_UP=nrow(subset(sig,log2FoldChange>0))
    N_DN=nrow(subset(sig,log2FoldChange<0))
    DET=nrow(de)
    HEADER=paste(N_SIG,"@5%FDR,", N_UP, "up", N_DN, "dn", DET, "detected")
    top <- head(sig,30)
    mylabels <- sapply(strsplit(rownames(top)," "),"[[",2)
    plot(de$log2FoldChange,-log10(de$pval),cex=0.5,pch=19,col="darkgray",
        main=name, xlab="log2 FC", ylab="-log10 pval")
    mtext(HEADER)
    grid()
    points(sig$log2FoldChange,-log10(sig$pval),cex=0.5,pch=19,col="red")
    text(top$log2FoldChange+0.2,-log10(top$pval),labels=mylabels, srt=35 ,cex=0.7)
}

make_heatmap <- function(de,name,myss,mx,n=30){
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  csc <- myss$Treatment
  csc <- gsub("C","orange",csc)
  csc <- gsub("T","gray",csc)
  mxn <- mx/rowSums(mx)*1000000
  x <- mxn[which(rownames(mxn) %in% rownames(head(de,n))),]
  heatmap.2(as.matrix(x),trace="none",col=colfunc(25),scale="row", margins = c(10,15), cexRow=0.7,
    main=paste("Top ranked",n,"genes in",name) , ColSideColors = csc  )
  mtext("ctrl=orange, trt=gray")
}

make_heatmap2 <- function(de,name,myss,mx,n=30){
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  csc <- myss$Treatment  
  csc <- gsub("C","orange",csc)
  csc <- gsub("T","gray",csc)
  mxn <- mx/rowSums(mx)*1000000
  x <- mxn[which(rownames(mxn) %in% rownames(head(de,n))),]
  rownames(x) <- sapply(strsplit(rownames(x)," "),"[[",2)
  heatmap.2(as.matrix(x),trace="none",col=colfunc(25),scale="row", margins = c(10,15), cexRow=0.6,
    main=paste("Top ranked",n,"genes in",name) ,  ColSideColors = csc   )
  mtext("ctrl=orange, trt=gray")
}

mymds <- function(de,name,myss,mx) {
  mds <- cmdscale(dist(t(mx)))
  csc <-  myss$Treatment
  csc <- gsub("C","orange",csc)
  csc <- gsub("T","gray",csc)
  plot(mds, xlab="Coordinate 1", ylab="Coordinate 2", main = name ,
    type = "p",bty="n",pch=19, cex=4 ,col=csc )
  text(mds, labels=rownames(mds) ,col="black")
  legend("topright", inset=.02, title="treatment",
    legend=unique(as.factor(ss$Treatment)) , pch=19, col=unique(csc),  cex=1.4)
}


# make plots for contrast 1
maplot(dge1,"Cont1: Effect of treatment in hypothalamus")
make_volcano(dge1,"Cont1: Effect of treatment in hypothalamus")
#make_volcano2(dge1,"Cont1: Effect of treatment in hypothalamus")
make_heatmap(de=dge1,name="Cont1: Effect of treatment in hypothalamus",myss=ss1,mx=xx1,n=50)
make_heatmap2(de=dge1,name="Cont1: Effect of treatment in hypothalamus",myss=ss1,mx=xx1,n=50)
mymds(de=dge1,name="Cont1: Effect of treatment in hypothalamus",myss=ss1,mx=xx1)

```

Now let's look at hippocampus.

```{r,contrast2}

dds <- DESeqDataSetFromMatrix(countData = xx2 , colData = ss2, design = ~ Treatment )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:20,1:6] %>% kbl(caption = "Top gene expression differences for contrast 2: Effect of treatment in hippocampus") %>%
  kable_paper("hover", full_width = F)
dge2 <- dge
d2up <- rownames(subset(dge2,padj <= 0.05 & log2FoldChange > 0))
d2dn <- rownames(subset(dge2,padj <= 0.05 & log2FoldChange < 0))
write.table(dge2,file="dge2.tsv",quote=FALSE,sep="\t")

maplot(dge2,"Cont2: Effect of treatment in hippocampus")
make_volcano(dge2,"Cont2: Effect of treatment in hippocampus")
make_heatmap(de=dge2,name="Cont2: Effect of treatment in hippocampus",myss=ss2,mx=xx2,n=50)
make_heatmap2(de=dge2,name="Cont2: Effect of treatment in hippocampus",myss=ss2,mx=xx2,n=50)
mymds(de=dge2,name="Cont2: Effect of treatment in hippocampus",myss=ss2,mx=xx2)

```

Now let's look at amygdala.

```{r,contrast3}

dds <- DESeqDataSetFromMatrix(countData = xx3 , colData = ss3, design = ~ Treatment )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:20,1:6] %>% kbl(caption = "Top gene expression differences for contrast 3: Effect of treatment in amygdala") %>%
  kable_paper("hover", full_width = F)
dge3 <- dge
d3up <- rownames(subset(dge3,padj <= 0.05 & log2FoldChange > 0))
d3dn <- rownames(subset(dge3,padj <= 0.05 & log2FoldChange < 0))
write.table(dge3,file="dge3.tsv",quote=FALSE,sep="\t")

maplot(dge3,"Cont3: Effect of treatment in amygdala")
make_volcano(dge3,"Cont3: Effect of treatment in amygdala")
make_heatmap(de=dge3,name="Cont3: Effect of treatment in amygdala",myss=ss3,mx=xx3,n=50)
make_heatmap2(de=dge3,name="Cont3: Effect of treatment in amygdala",myss=ss3,mx=xx3,n=50)
mymds(de=dge3,name="Cont3: Effect of treatment in amygdala",myss=ss3,mx=xx3)

```

Now let's look at periaqueductal gray.

```{r,contrast4}

dds <- DESeqDataSetFromMatrix(countData = xx4 , colData = ss4, design = ~ Treatment )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:20,1:6] %>% kbl(caption = "Top gene expression differences for contrast 4: Effect of treatment in periaqueductal gray") %>%
  kable_paper("hover", full_width = F)
dge4 <- dge
d4up <- rownames(subset(dge4,padj <= 0.05 & log2FoldChange > 0))
d4dn <- rownames(subset(dge4,padj <= 0.05 & log2FoldChange < 0))
write.table(dge4,file="dge4.tsv",quote=FALSE,sep="\t")

maplot(dge4,"Cont4: Effect of treatment in PAG")
make_volcano(dge4,"Cont4: Effect of treatment in PAG")
make_heatmap(de=dge4,name="Cont4: Effect of treatment in PAG",myss=ss4,mx=xx4,n=50)
make_heatmap2(de=dge4,name="Cont4: Effect of treatment in PAG",myss=ss4,mx=xx4,n=50)
mymds(de=dge4,name="Cont4: Effect of treatment in PAG",myss=ss4,mx=xx4)

```

Now let's look at nucleus incertus.

```{r,contrast5}

dds <- DESeqDataSetFromMatrix(countData = xx5 , colData = ss5, design = ~ Treatment )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:20,1:6] %>% kbl(caption = "Top gene expression differences for contrast 5: Effect of treatment in nucleus incertus") %>%
  kable_paper("hover", full_width = F)
dge5 <- dge
d5up <- rownames(subset(dge5,padj <= 0.05 & log2FoldChange > 0))
d5dn <- rownames(subset(dge5,padj <= 0.05 & log2FoldChange < 0))
write.table(dge5,file="dge5.tsv",quote=FALSE,sep="\t")

maplot(dge5,"Cont5: Effect of treatment in NI")
make_volcano(dge5,"Cont5: Effect of treatment in NI")
make_heatmap(de=dge5,name="Cont5: Effect of treatment in NI",myss=ss5,mx=xx5,n=50)
make_heatmap2(de=dge5,name="Cont5: Effect of treatment in NI",myss=ss5,mx=xx5,n=50)
mymds(de=dge5,name="Cont5: Effect of treatment in NI",myss=ss5,mx=xx5)

```








## Single contrast pathway analysis with mitch

Firstly need to conduct mitch enrichment analysis for each contrast separately.

```{r,mitch1}

if  (!file.exists("mouse_msigdb_reactome_2022-02-16.gmt")) {
  download.file("http://ziemann-lab.net/public/msigdb_mouse/mouse_msigdb_reactome_2022-02-16.gmt",
    destfile="mouse_msigdb_reactome_2022-02-16.gmt")
}
genesets <- gmt_import("mouse_msigdb_reactome_2022-02-16.gmt")
names(genesets) <- gsub("REACTOME_","",names(genesets))
names(genesets) <- gsub("_"," ",names(genesets))

# gene table
gt <- as.data.frame(rownames(xx))
gt$gene <- sapply(strsplit(gt[,1]," "),"[[",2)


```

```{r,mitch2}

# contrast1
m1 <- mitch_import(dge1, DEtype="deseq2",geneTable=gt)
mres1 <- mitch_calc(m1, genesets, priority="effect")
head(mres1$enrichment_result,20) %>% kbl(caption = "Top gene pathway differences in contrast 1") %>% kable_paper("hover", full_width = F)
m1top <- subset(mres1$enrichment_result,p.adjustANOVA<0.05)
m1up <- subset(m1top,s.dist>0)$set
m1dn <- subset(m1top,s.dist<0)$set
mitch_report(mres1,outfile="mitch1.html",overwrite=TRUE)
write.table(mres1$enrichment_result,file="mitch1.tsv",quote=FALSE,sep="\t",row.names=FALSE)

m1top_up <- head(subset(m1top,s.dist>0),10)[,"s.dist"]
names(m1top_up) <- head(subset(m1top,s.dist>0),10)[,"set"]
m1top_dn <- head(subset(m1top,s.dist<0),10)[,"s.dist"]
names(m1top_dn) <- head(subset(m1top,s.dist<0),10)[,"set"]
m1top_updn <- c(m1top_up,m1top_dn)
m1top_updn <- m1top_updn[order(m1top_updn)]

par(mar=c(5,25,3,1))
barplot(m1top_updn,horiz=TRUE,las=1,col="darkgray",
  xlab="Enrichment score",cex.names=0.8,xlim=c(-1,1),
  main="Pathway changes in hypothalamus")
grid()

# contrast2
m2 <- mitch_import(dge2, DEtype="deseq2",geneTable=gt)
mres2 <- mitch_calc(m2, genesets, priority="effect")
head(mres2$enrichment_result,20) %>% kbl(caption = "Top gene pathway differences in contrast 2") %>% kable_paper("hover", full_width = F)
m2top <- subset(mres2$enrichment_result,p.adjustANOVA<0.05)
m2up <- subset(m2top,s.dist>0)$set
m2dn <- subset(m2top,s.dist<0)$set
mitch_report(mres2,outfile="mitch2.html",overwrite=TRUE)
write.table(mres2$enrichment_result,file="mitch2.tsv",quote=FALSE,sep="\t",row.names=FALSE)

m2top_up <- head(subset(m2top,s.dist>0),10)[,"s.dist"]
names(m2top_up) <- head(subset(m2top,s.dist>0),10)[,"set"]
m2top_dn <- head(subset(m2top,s.dist<0),10)[,"s.dist"]
names(m2top_dn) <- head(subset(m2top,s.dist<0),10)[,"set"]
m2top_updn <- c(m2top_up,m2top_dn)
m2top_updn <- m2top_updn[order(m2top_updn)]

par(mar=c(5,25,3,1))
barplot(m2top_updn,horiz=TRUE,las=1,col="darkgray",
  xlab="Enrichment score",cex.names=0.8,xlim=c(-1,1), 
  main="Pathway changes in hippocampus")
grid()

# contrast3
m3 <- mitch_import(dge3, DEtype="deseq2",geneTable=gt)
mres3 <- mitch_calc(m3, genesets, priority="effect")
head(mres3$enrichment_result,20) %>% kbl(caption = "Top gene pathway differences in contrast 3") %>% kable_paper("hover", full_width = F)
m3top <- subset(mres3$enrichment_result,p.adjustANOVA<0.05)
m3up <- subset(m3top,s.dist>0)$set
m3dn <- subset(m3top,s.dist<0)$set
mitch_report(mres3,outfile="mitch3.html",overwrite=TRUE)
write.table(mres3$enrichment_result,file="mitch3.tsv",quote=FALSE,sep="\t",row.names=FALSE)

m3top_up <- head(subset(m3top,s.dist>0),10)[,"s.dist"]
names(m3top_up) <- head(subset(m3top,s.dist>0),10)[,"set"]
m3top_dn <- head(subset(m3top,s.dist<0),10)[,"s.dist"]
names(m3top_dn) <- head(subset(m3top,s.dist<0),10)[,"set"]
m3top_updn <- c(m3top_up,m3top_dn)
m3top_updn <- m3top_updn[order(m3top_updn)]

par(mar=c(5,25,3,1))
barplot(m3top_updn,horiz=TRUE,las=1,col="darkgray",
  xlab="Enrichment score",cex.names=0.8,xlim=c(-1,1),
  main="Pathway changes in amygdala")
grid()

# contrast4
m4 <- mitch_import(dge4, DEtype="deseq2",geneTable=gt)
mres4 <- mitch_calc(m4, genesets, priority="effect")
head(mres4$enrichment_result,20) %>% kbl(caption = "Top gene pathway differences in contrast 4") %>% kable_paper("hover", full_width = F)
m4top <- subset(mres4$enrichment_result,p.adjustANOVA<0.05)
m43up <- subset(m4top,s.dist>0)$set
m4dn <- subset(m4top,s.dist<0)$set
mitch_report(mres4,outfile="mitch4.html",overwrite=TRUE)
write.table(mres4$enrichment_result,file="mitch4.tsv",quote=FALSE,sep="\t",row.names=FALSE)

m4top_up <- head(subset(m4top,s.dist>0),10)[,"s.dist"]
names(m4top_up) <- head(subset(m4top,s.dist>0),10)[,"set"]
m4top_dn <- head(subset(m4top,s.dist<0),10)[,"s.dist"]
names(m4top_dn) <- head(subset(m4top,s.dist<0),10)[,"set"]
m4top_updn <- c(m4top_up,m4top_dn)
m4top_updn <- m4top_updn[order(m4top_updn)]

par(mar=c(5,25,3,1))
barplot(m4top_updn,horiz=TRUE,las=1,col="darkgray",
  xlab="Enrichment score",cex.names=0.8,xlim=c(-1,1),
  main="Pathway changes in periaqueductal gray")
grid()

# contrast5
m5 <- mitch_import(dge5, DEtype="deseq2",geneTable=gt)
mres5 <- mitch_calc(m5, genesets, priority="effect")
head(mres5$enrichment_result,20) %>% kbl(caption = "Top gene pathway differences in contrast 5") %>% kable_paper("hover", full_width = F)
m5top <- subset(mres5$enrichment_result,p.adjustANOVA<0.05)
m5up <- subset(m5top,s.dist>0)$set
m5dn <- subset(m5top,s.dist<0)$set
mitch_report(mres5,outfile="mitch5.html",overwrite=TRUE)
write.table(mres5$enrichment_result,file="mitch5.tsv",quote=FALSE,sep="\t",row.names=FALSE)

m5top_up <- head(subset(m5top,s.dist>0),10)[,"s.dist"]
names(m5top_up) <- head(subset(m5top,s.dist>0),10)[,"set"]
m5top_dn <- head(subset(m5top,s.dist<0),10)[,"s.dist"]
names(m5top_dn) <- head(subset(m5top,s.dist<0),10)[,"set"]
m5top_updn <- c(m5top_up,m5top_dn)
m5top_updn <- m5top_updn[order(m5top_updn)]

par(mar=c(5,25,3,1))
barplot(m5top_updn,horiz=TRUE,las=1,col="darkgray",
  xlab="Enrichment score",cex.names=0.8,xlim=c(-1,1),
  main="Pathway changes in nucleus incertus")
grid()

```

## Conclusion

There was some variability among samples in the hypothalamus, resulting in no statistically significant findings.
However the heatmap does show some potentially interesting patterns.
The gene pathway analysis did show some differences.
The mitch software can perform multi-contrast pathway analysis and will be important to contrast the effect of the treatment
in different tissues.

## References

Bibliography

1.	Babraham bioinformatics - FastQC A quality control tool for high throughput sequence data. Babraham.ac.uk, https://www.bioinformatics.babraham.ac.uk/projects/fastqc/ (accessed February 23, 2022).

2.	Frankish A, Diekhans M, Jungreis I, et al. GENCODE 2021. Nucleic Acids Res 2021; 49: D916–D923.

3.	Jiang H, Lei R, Ding S-W, et al. Skewer: a fast and accurate adapter trimmer for next-generation sequencing paired-end reads. BMC Bioinformatics 2014; 15: 182.

4.	Bray NL, Pimentel H, Melsted P, et al. Near-optimal probabilistic RNA-seq quantification. Nat Biotechnol 2016; 34: 525–527.

5.	Ewels P, Magnusson M, Lundin S, et al. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics 2016; 32: 3047–3048.

6.	Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol 2014; 15: 550.

7.	Liberzon A, Birger C, Thorvaldsdóttir H, et al. The Molecular Signatures Database (MSigDB) hallmark gene set collection. Cell Syst 2015; 1: 417–425.

8.	Jassal B, Matthews L, Viteri G, et al. The reactome pathway knowledgebase. Nucleic Acids Res 2020; 48: D498–D503.

9.	Dolgalev I. MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format [R package msigdbr version 7.4.1], https://cran.r-project.org/web/packages/msigdbr/index.html (2021, accessed February 23, 2022).

10.	Kaspi A, Ziemann M. Mitch: Multi-contrast pathway enrichment for multi-omics and single-cell profiling data. BMC Genomics 2020; 21: 447.

## Session information

For reproducibility.

```{r,sessioninfo}

sessionInfo()

```
